terraform/modules/testing/rbac_test_lambda_files/rbac_lambda.py

```diff
@@ -97,41 +96,44 @@ def use_database(session_url, database_name):
-def check_for_access_denied(session_url, pii_table_name):
-    print("Attempting to access PII data")
-    statements_url = session_url + '/statements'
-    code = {'code': f'spark.sql("select * from {pii_table_name}")'}
-    status_url = initial_request(statements_url, code)
-    response = poll_for_result(session_url, status_url)
-    if access_denied_message in response['output']['evalue']:
-        print("Expected 403 - Got 403")
-        return "OK"
-    elif response['output']['status'] == "error":
-        kill_session(session_url)
-        print(response['output']['evalue'])
-        sys.exit('Expected 403 - But received a different error')
-    else:
-        kill_session(session_url)
-        print(response['output']['evalue'])
-        sys.exit('Expected 403 - But did not receive access denied')
-
-
-# Attempt to access a table with the pii:false tag - should be successful
-def check_for_access_granted(session_url, non_pii_table_name):
-    print("Attempting to access Non PII data")
+def check_access_is_correct(session_url, table, access):
     statements_url = session_url + '/statements'
-    code = {'code': f'spark.sql("select * from {non_pii_table_name}")'}
+    code = {'code': f'spark.sql("select * from {table["name"]}")'}
     status_url = initial_request(statements_url, code)
     response = poll_for_result(session_url, status_url)
     print(response)
-    if response['output']['status'] != "error":
-        print("Expected no error - No error found")
-        return "OK"
-    else:
-        kill_session(session_url)
-        print(response['output']['evalue'])
-        sys.exit('Expected data - Received an error')
+
+    # Should have full access to all data
+    if access == "pii":
+        # Error - exit and print error
+        if response['output']['status'] == "error":
+            kill_session(session_url)
+            print(response['output']['evalue'])
+            sys.exit('Expected data - Received error')
+        # Received data as expected
+        else:
+            kill_session(session_url)
+            print("Expected data - Received data")
+            return "OK"
+
+    # Should only have access to non-pii data
+    if access == "non_pii":
+        # Should receive 403 when trying to access pii data
+        if table["type"] == "pii" and access_denied_message in response['output']['evalue']:
+            kill_session(session_url)
+            print("Expected 403 - received 403")
+            return "OK"
+        # Error - exit and print error
+        elif response['output']['status'] == "error":
+            kill_session(session_url)
+            print(response['output']['evalue'])
+            sys.exit('Expected data - Received error')
+        # Finally, if no error, then non-pii data was returned as expected
+        else:
+            kill_session(session_url)
+            print("Expected data - Received data")
+            return "OK"
```

Comment:
Wouldn't the else block be triggered under these conditions?
 * `table["type"] == "pii"`
 * `acess_denied_message not in response[..][..]`
 *  `response['output']['status'] != "error"`
Which to me means a non-pii access user, trying to access a pii table, no access_denied_message present in the response, and no error therefore pii data received.

Reply:
Great catch - that's the exact scenario we want to test for. Somehow I missed it during this refactor. 
Have amended the code now.